sonic-net/sonic-utilities
# Add Transceiver PM basic CLI support to show output from TRANSCEIVER_PM table for ZR

#### What I did
Add Transceiver Performance Monitoring CLI support to show output from TRANSCEIVER_PM table for ZR optics. This is basic version CLI, it just displays the pm info available in current xcvrd/xcvr_api/DB infra, no bucketization, no history.

#### How I did it
1. Add pm support in sfpshow related scripts. 
2. Add pytest in sfp_test.py and mock DB info. 

#### How to verify it
Run 'show interface transceiver pm' with or without specifying interface name:
1. Verify CLI can properly display output for all interfaces in one shot and one particular interface.
2. For interfaces that don't have pm info, verify the output is N/A.
3. For interfaces that have pm info, verify proper output is displayed.
4. Verify the pm info displayed in CLI matches with DB .

#### Previous command output (if the output of a command-line utility has changed)
N/A

#### New command output (if the output of a command-line utility has changed)

```
root@sonic:/home/cisco# show interface transceiver pm                                                                                 
Ethernet0: Transceiver performance monitoring Not applicable                                                                                                                                              
                                                                                                                                                                                                          
Ethernet8: Transceiver performance monitoring Not applicable                                                                                                                                              
                                                                                                                                                                                                          
Ethernet16: Transceiver performance monitoring Not applicable
......
Ethernet272: Transceiver performance monitoring applicable
    Parameter        Unit              Min            Avg            Max  Threshold    Threshold             Threshold     Threshold    Threshold    Threshold
                                                                          High         High                  Crossing      Low          Low          Crossing
                                                                          Alarm        Warning               Alart-High    Alarm        Warning      Alart-Low
    ---------------  ------  -------------  -------------  -------------  -----------  --------------------  ------------  -----------  -----------  -----------
    Tx Power         dBm      -8.22          -8.23          -8.24         -5.0         -6.0                  False         -16.99       -16.003      False
    Rx Total Power   dBm     -10.61         -10.62         -10.62         2.0          0.0                   False         -21.0        -18.0        False
    Rx Signal Power  dBm     -40              0             40            13.0         10.0                  True          -18.0        -15.0        True
    PDL              dB        0.5            0.6            0.6          4.0          4.0                   False         0.0          0.0          False
    OSNR             dB       36.5           36.5           36.5          99.0         99.0                  False         0.0          0.0          False
    eSNR             dB       30.5           30.5           30.5          99.0         99.0                  False         0.0          0.0          False
    CFO              MHz      54             70            121            3800.0       3800.0                False         -3800.0      -3800.0      False
    DGD              ps        5.37           5.56           5.81         7.0          7.0                   False         0.0          0.0          False
    SOPMD            ps^2      0              0              0            655.35       655.35                False         0.0          0.0          False
    SOP ROC          krad/s    1              1              2            N/A          N/A                   N/A           N/A          N/A          N/A
    Pre-FEC BER      N/A       0.000457501    0.000465781    0.000575639  0.0125       0.011000000000000001  False         0.0          0.0          False
    Post-FEC BER     N/A       0              0              0            1000.0       1.0                   False         0.0          0.0          False
    EVM              %       100            100            100            N/A          N/A                   N/A           N/A          N/A          N/A
Ethernet280: Transceiver performance monitoring Not applicable
     
```
